server/routes/api.js

```diff
@@ -2,8 +2,125 @@ const express = require('express');
 
 const router = express.Router();
 
-router.get('/', (req, res) => {
-  res.status(200).json({ message: 'hi' });
-});
+/**
+ * When it is triggered: should be constantly checked on interval by logged in user
+ * What it does: checks for and returns any updates to open parties associated with current user; 
+ *    also returns any closed parties with a date from the last 24hours;
+ * sample request data from frontend: just the req.params.userid
+ * sample response data after controllers:
+ * const data = [
+   {
+    party: {
+      _id: 12,
+      date: 'date string that JS can parse',
+      host: 'Derek',
+      hostVenmo: derek1818,
+      name: 'drunk brunch',
+      menu: 'http://www.brunchplace.com/menu',
+      status: 'open',
+    },
+    orders: [
+      {
+        name: 'Derek',
+        status: submitted,
+        tax_amount: 200,
+        tip_ammont: 150,
+        items: [
+          { name: 'burger', cost: 950, quantity: 2 },
+          { name: 'fries', cost: 150, quantity: 1 },
+          { name: 'coke', cost: 150, quantity: 1 },
+        ],
+      },
+      {
+        name: 'Kyle',
+        status: submitted,
+        tax_amount: 170,
+        tip_ammont: 250,
+        items: [
+          { name: 'sandwich', cost: 850, quantity: 1 },
+          { name: 'shake', cost: 550, quantity: 1 },
+        ],
+      },
+      {
+        name: 'Joe',
+        status: open,
+      },
+    ],
+  },
+  {
+    party: 'a possible second open party',
+    orders: ['another orders array'],
+  },
+]
+ *
+ */
+router.get(
+  '/party/:userid',
+  /*TODO: add middleware */ (req, res) => {
+    res.status(200).json(res.locals);
+  }
+);
+
+/**
+ * When it is triggered: when a user creates a new party
+ * What it does: creates a new open party row in DB and creates a new open order row for each guest/host
+ * sample request data from frontend:
+ * const data = {
+  date: 'date string that JS can parse',
+  host_name: 'Derek',
+  host_id: 1,
+  name: 'drunk brunch',
+  menu: 'http://www.brunchplace.com/menu',
+  guests: ['Joe', 'Kyle', 'Katy'],
+};  
```

Comment:
I might be missing something, but can you remind me how we're planning to map the guests to users in the db if we only get their first names from the front end?

Reply:
so that guests array will be the usernames that the host entered.  So at least for MVPs we will assume that the host only enter valid usernames that we have already stored in our database.  I made it confusing by making the usernames look too much like first names.

Reply:
Ah, that makes sense. Thanks!